opentuner/search/composableevolutionarytechniques.py

```diff
@@ -359,19 +395,68 @@ def select_parameters(self, params):
     """
     randomly select a subset of parameters to operate on
     """
-    ret_list = []
     random.shuffle(params)
-    for i, k in enumerate(params):
-      if i < self.n_cross or random.random() < self.cr:
+    ret_list = params[:self.must_mutate_count]
+    for param in params[self.must_mutate_count:]:
+      if random.random() < self.cr:
         ret_list.append(k)
     return ret_list
 
   def get_default_operator(self, param_type):
     return {'op_name': 'op4_set_linear', 'args': [1.0, self.use_f, -self.use_f], 'kwargs': {}}
 
+class GreedyComposableTechnique(ComposableEvolutionaryTechnique):
+  """
+  Always mixes in global best as parents
+  """
+  def __init__(self,
+               mutation_rate = 0.1,
+               must_mutate_count = 1,
+               *pargs, **kwargs):
+    super(GreedyComposableTechnique, self).__init__(*pargs, **kwargs)
+    self.mutation_rate = mutation_rate
+    self.must_mutate_count = must_mutate_count
+    self.population_size = 10
+
+  def minimum_number_of_parents(self):
+    return 4
+
+  def get_parents(self, population):
+    population.sort(key=_.timestamp) # sort population by timestamp
+
+    # mix in base with best as many operators do nothing given identical input cfgs
+    cfg = self.manipulator.copy(population[0].config)
```

Comment:
Should be okay. Some operators will still generate a new config. Otherwise, we'll return a duplicate configuration, but because this is a sequential search technique, it will "retry" and do another iteration on the next population member.